src/cmd/vm.rs

```diff
@@ -20,136 +27,231 @@ pub struct Args {
+    extra_args: Option<String>,
 }
 
-fn run_connect(_args: &ArgsConnect) -> Result<()> {
+fn run_setup(args: &ArgsSetup) -> Result<()> {
+    let dir = find_betty_script(&args.arc)?;
+
+    println!("Updating packages...");
+    let update_package = Command::new("sudo")
+        .args(["apt", "update"])
+        .spawn()
+        .context("Failed to execute sudo apt update")?;
+    update_package
+        .wait_with_output()
+        .context("Failed to wait for updating packages")?;
+
+    println!("Enable KVM...");
+    enable_kvm()?;
+
+    println!("Installing python packages...");
+    let install_python_package = Command::new("sudo")
+        .args(["apt", "install", "python3-pip", "python3-venv"])
+        .spawn()
+        .context("Failed to install python packages")?;
+    install_python_package
+        .wait_with_output()
+        .context("Failed to wait for installing python packages")?;
+
+    println!("Running betty.sh setup...");
+    let arg = match &args.extra_args {
+        Some(a) => a,
+        None => "",
+    };
+    run_betty(&dir, "setup", arg)?;
+
+    println!("Running gcloud auth login...");
+    let gcloud_auth = Command::new("gcloud")
+        .args(["auth", "login"])
+        .spawn()
+        .context("Failed to run gcloud login gcloud")?;
+    gcloud_auth
+        .wait_with_output()
+        .context("Failed to wait for gcloud auth login")?;
+
     Ok(())
 }
 
-#[derive(FromArgs, PartialEq, Debug)]
-/// run first time setup, installs necessary dependencies
-#[argh(subcommand, name = "setup")]
-pub struct ArgsSetup {
-    /// extra arguments
-    #[argh(option)]
-    extra_args: Vec<String>,
-}
+fn enable_kvm() -> Result<()> {
+    let username = whoami::username();
+
+    println!("Installing kvm support...");
+    let install_kvm_support = Command::new("sudo")
+        .args(["apt-get", "install", "qemu-system-x86"])
+        .spawn()
+        .context("Failed to install kvm support")?;
+    install_kvm_support
+        .wait_with_output()
+        .context("Failed to wait for installing kvm support")?;
+
+    println!("Loading Kernel modules...");
+    let load_kernel_module = Command::new("sudo")
+        .args(["modprobe", "kvm-intel"])
+        .spawn()
+        .context("Failed to load kernel modules")?;
+    load_kernel_module
+        .wait_with_output()
+        .context("Failed to wait for loading kernel modules")?;
+
+    println!("Adding the user to the kvm local group...");
+    let add_to_kvm_group = Command::new("sudo")
+        .args(["adduser", &username, "kvm"])
+        .spawn()
+        .context("Failed to add the user to the kvm local group")?;
+    add_to_kvm_group
+        .wait_with_output()
+        .context("Failed to wait for adding the user to the kvm local group")?;
+
+    let is_kvm_enable = run_bash_command(
+        "[[ -e /dev/kvm ]] && grep '^flags' /proc/cpuinfo | grep -qE 'vmx|svm'",
+        None,
+    )?;
+    if !is_kvm_enable.status.success() {
```

Comment:
I think .status.success should be moved to line 138 here inlined so is_kvm_enabled is just a bool